docs/how-to/builtins.rst

```diff
@@ -1,104 +1,138 @@
-handles running your code in both interpreters and comparing outputs.
+        Note also the format for errors: ``throw new exceptions.<Error>.$pyclass``.
 
-Let's look at some test code that checks if a the Batavia implementation of ``bool`` can handle a
-bool-like class that implements ``__bool__``.
+Returning a value
+*****************
 
-.. code-block:: Python
+Builtins implement Python functions and should return a Python object.
+Batavia implementations of all Python types are located in ``/batavia/types.js``.
+JavaScript imports use the ``require`` keyword and can be imported inline or at 
+the top of the file. Inline imports can be preferable in some cases.
 
-    def test_bool_like(self):
-        self.assertCodeExecution("""
-            class BoolLike:
-                def __init__(self, val):
-                    self.val = val
+.. code-block:: javascript
 
-                def __bool__(self):
-                    return self.val == 1
-            print(bool(BoolLike(0)))
-            print(bool(BoolLike(1)))
-            """)
+    ...
 
-The ``assertCodeExecution`` method will run the code provided to it in both implementations. This
-code needs to generate some output so that the output can be compared, hence the need to print the
-values.
+    Tuple = require('../types.js').Tuple
+    return new Tuple(my, results, here)
+    }
 
+Documentation
+*************
 
-Process
-----------
+Finally, add the docstring to the function object. In JavaScript, like in Python, functions 
+are first-class objects and can have additional properties.
+
+.. code-block:: javascript
+
+    <fn>.__doc__ = 'docstring from Python 3.x goes here, for documentation'
+
+    modules.export = <fn>
 
-For a given function, run `functionname.__doc__` in the Python 3.4 repl
+Tests
+*****
 
-Copy the docstring into the doc
+No implemenation for a project like this is complete without tests. Check out the other sections for
+more details on test structure. Tests are located in ``/tests`` in a similar folder structure to the
+core code, and most test files have already been created. Some things that should almost always be
+tested:
 
-Run the function in Python 3.4
+* Write a test or three to ensure your function returns the correct output with some normal inputs.
+* Think of a few weird inputs that could throw off your code (or future code). Test them.
+* If you are throwing an error (excluding ``BataviaError``) anywhere, write a test that tries to throw it.
+* If you accounted for an edge case (look for an ``if`` statement), test it.
+* Check out the official documentation (https://docs.python.org/3/) for more edge cases.
 
-Take a guess at the implementation structure based on the other functions.
+Credit
+^^^^^^
 
-Copy the style of the other implemented functions
+Thanks to Kshitij Sobti for the original document that this guide is based on. Most code examples are from his original.
```

Comment:
I appreciate the sentiment here of recognizing past authors, but it's a little out of place in main project documentation.